code/modules/mob/living/carbon/human/species/station/prometheans.dm

```diff
@@ -179,13 +179,15 @@ var/datum/species/shapeshifter/promethean/prometheans
 	if(istype(T))
 		if(!(H.shoes || (H.wear_suit && (H.wear_suit.body_parts_covered & FEET))))
 			for(var/obj/O in T)
-				O.clean_blood()
-				H.adjust_nutrition(rand(5, 15))
+				if(O.clean_blood())
+					H.adjust_nutrition(rand(5, 15))
 			if (istype(T, /turf/simulated))
 				var/turf/simulated/S = T
-				T.clean_blood()
-				S.dirt = 0
-				H.adjust_nutrition(rand(10, 20))
+				if(T.clean_blood())
+					H.adjust_nutrition(rand(10, 20))
+				if(S.dirt > 50)
```

Comment:
Just an idle suggestion:
````
if(S.dirt)
  var/nutrition = min(S.dirt, 50) // or some other desirable cap
  S.dirt -= nutrition
  H.adjust_nutrition(max(round(nutrition/20), 1) 
````
This would make amount of dirt determine how much it feeds them, and would stop them eating 500u of dirt in one tick.